src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -203,11 +210,9 @@ public Compilation Clone()
         /// True if the SemanticModel should ignore accessibility rules when answering semantic questions.
         /// </param>
         public SemanticModel GetSemanticModel(SyntaxTree syntaxTree, bool ignoreAccessibility = false)
-        {
-            return CommonGetSemanticModel(syntaxTree, ignoreAccessibility);
-        }
+            => GetSemanticModelCore(syntaxTree, ignoreAccessibility, useSemanticModelProviderIfNonNull: true);
 
-        protected abstract SemanticModel CommonGetSemanticModel(SyntaxTree syntaxTree, bool ignoreAccessibility);
+        internal abstract SemanticModel GetSemanticModelCore(SyntaxTree syntaxTree, bool ignoreAccessibility, bool useSemanticModelProviderIfNonNull);
```

Comment:
Think it would be clearer to have two methods here:

1. `GetSemanticModelCore`: this gets a `SemanticModel` which is potentially a cached instance. It's at the discretion of the `Compilation` to make the decision on how it is created and returned
1. `CreateSemanticModelCore`: this always creates a new `SemanticModel`. I find that is much clearer than calling `Get` and using a flag `useSemanticModelProviderIfNonNull: false`.

Reply:
Ah, that is much better, will split.